src/hocs/withFeatureFlag.tsx

```diff
@@ -1,23 +1,26 @@
 "use client";
 import React, { useEffect } from "react";
 import { useRouter } from "next/navigation";
-import { useFeatureFlagEnabled } from "posthog-js/react";
+import { useFeatureFlagVariantKey } from "posthog-js/react";
 
 const withFeatureFlag = <P extends object>(
   WrappedComponent: React.ComponentType<P>,
   featureFlag: string,
+  featureFlagVariant: string | true = true,
```

Comment:
Needed this hoc to support non-boolean feature flags (I.e. those with more than 2 variants). So it can now accept an optional string to match.